### ISSUE TITLE
Remove character limit on matrix survey option type (and switch to JSONSchemaForm)

### ISSUE THREAD

**Issue by avaidyam** [BIDMCDigitalPsychiatry/LAMP-platform#485]
There is not supposed to be a character limit for any survey question types - and we should be using the JSON Schema Form system for the clinician user interface to create surveys.

**Comment by sarithapillai8** [comment]
@avaidyam 
Before starting with this change, just confirming that there will be design issues if we remove character limit for all the question types. It will be v=based on the input given by the user. For eg: if long texts are provided for slider, rating etc.

**Comment by avaidyam** [comment]
@sarithapillai8 Can you share where the character limit was requested? I don't recall us wanting to limit it, only that we wanted the text to wrap correctly to avoid spilling over the edge of the screen, and the page scrolling down more.

**Comment by sarithapillai8** [comment]
@avaidyam No I was not saying about the character limit. I was saying about jsonschema for survey activity which was put on hold. So please consider that as feature change, not bug.
I just noted the the design issues that may arise if we remove the character limit. With scrolling also it may not look nice, And normally we adds character limits to fields in a form. We can remove it, It can be considered as a bug.

**Comment by avaidyam** [comment]
>  I was saying about jsonschema for survey activity which was put on hold.

I don't recall it being put on hold. [The initial changes for this were made back in March 2021.](https://github.com/BIDMCDigitalPsychiatry/LAMP-dashboard/commit/94e74bcc72dc741bfb16138147fbb05f0d155f90#diff-bc5d2f8f270abce0ee7fb8d0d317ba6bfe41fca9709e3ab8b2eb883ae90e4ceb) It was clear that we wanted to move all activity configuration to entirely be JSON Schema Form-driven. Let's make this change ASAP. 

> I just noted the the design issues that may arise if we remove the character limit. 

So let's fix the design issues then? We shouldn't penalize users for bad UI design/implementation. 

> With scrolling also it may not look nice

Vertical scrolling should be fine, horizontal scrolling is not good. 

> And normally we adds character limits to fields in a form

`Normally`, sure, but why? There's no point, even for normal websites and software. In reality, a user rarely wants to be limited by the software - especially mental health software designed for patients with serious mental illness.

> We can remove it, It can be considered as a bug.

👍

**Comment by sarithapillai8** [comment]
@avaidyam 
I don't have have the task where we discussed to put that on hold. But it was put on hold as you said for other urgent tasks as the survey would take much time to convert. Anyways we can start with that change.

**Comment by avaidyam** [comment]
> But it was put on hold as you said for other urgent tasks as the survey would take much time to convert. 

Sure - I think the benefit of moving forward with this change outweighs the development cost (i.e. bugs/feature requests) in the long-term. Please go ahead with it! 👍

**Comment by avaidyam** [comment]
It looks like the original issue that we placed `on hold` is #204. I'll close that issue since discussion has moved here. 

> **Reorder surveys in groups and questions within surveys**
> Currently, all non-Survey Activities have the JSONSchema editor which supports more advanced editing features such as reordering items within lists, and so on. The Survey Activity should also use the JSONSchema editor (no custom editor code for any Activities).

**Comment by sarithapillai8** [comment]
@avaidyam 
This is updated in dashboard-staging. QA completed.
